### PR TITLE
Add CI release workflow

### PR DESCRIPTION
**Pull request by jdrew82**
This PR is a starting point for adding a release workflow using GitHub Actions. The idea is to only start the release workflow when a tagged PR is created. As we can't guarantee that a build has already been done when this workflow is triggered I've included a standard environment build step prior to the package files being generated. It should then create the Release in GitHub when the tag is found and uses the files that are generated by poetry in the `dist` folder. Finally, it should push up to PyPI.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+name: Publish release to GitHub and push package to PyPI
+on:
+  pull_request:
+    branches: [main]
+    tags:
+      - "v*.*.*"
+
+jobs:
+  build:
```
Comment (chadell) [inline .github/workflows/release.yml]: on this release process, if I'm not wrong, we are just building and releasing, not actually testing.
Maybe we could use [workflow_run](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#workflow_run) to make sure that the release process happens after the test?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        if: startsWith(github.ref, 'refs/tags/')
+        with:
+          files: ./dist/*
+
+  push_to_registry:
+    name: Push Package to PyPI
+    runs-on: ubuntu-latest
+    steps:
+      - name: Push to PyPI
+        uses: pypa/gh-action/pypi-publish@release/v1
```
Comment (chadell) [inline .github/workflows/release.yml]:
```suggestion
        uses: pypa/gh-action-pypi-publish@release/v1
```

**Comment by jvanderaa** [comment]
This one has a few more stars than the other publishing to GH action.

https://github.com/marketplace/actions/upload-files-to-a-github-release

**Comment by jvanderaa** [comment]
Looks nice

**Pull request merged**